src/esp/bindings/AttributesBindings.cpp

```diff
@@ -229,16 +234,197 @@ void initAttributesBindings(py::module& m) {
+          each keyed by MarkerSet name for the particular link with the value being a list
+          of the marker points)");
+  py::class_<MarkerSets, esp::core::config::Configuration, MarkerSets::ptr>(
+      m, "MarkerSets")
+      .def(py::init(&MarkerSets::create<>))
+      .def_property_readonly(
+          "num_tasksets", &MarkerSets::getNumTaskSets,
+          R"(The current number of TaskSets present in the MarkerSets collection.)")
+      .def("has_taskset", &MarkerSets::hasTaskSet,
+           R"(Whether or not a TaskSet with the given name exists)",
+           "taskset_name"_a)
+      .def("has_task_linkset", &MarkerSets::hasTaskLinkSet,
+           R"(Whether or not a LinkSet with the given name within the TaskSet
+          with the given name exists)",
+           "taskset_name"_a, "linkset_name"_a)
+      .def("has_task_link_markerset", &MarkerSets::hasTaskLinkMarkerSet,
+           R"(Whether or not a MarkerSet exists within an existing Linkset in
+          an existing TaskSet with the given names)",
+           "taskset_name"_a, "linkset_name"_a, "markerset_name"_a)
+      .def(
+          "init_task_link_markerset", &MarkerSets::initTaskLinkMarkerSet,
+          R"(Initialize a MarkerSet within a LinkSet within a new TaskSet with the given
+           names in this collection)",
+          "taskset_name"_a, "linkset_name"_a, "markerset_name"_a)
+      .def("get_all_taskset_names", &MarkerSets::getAllTaskSetNames,
+           R"(Get a list of all the existing TaskSet names)")
+      .def("get_taskset", &MarkerSets::editTaskSet,
+           R"(Get an editable reference to the specified TaskSet, possibly
+            empty if it does not exist)",
+           "taskset_name"_a)
+      .def(
+          "set_taskset_points", &MarkerSets::setTaskSetPoints,
+          R"(Set all the marker points in the specified TaskSet to the 3d point values in the
+          passed dict of dicts. The format should be a dictionary keyed by LinkSet name, of
+          dictionaries, each keyed by MarkerSet name and referencing a list of 3d points)",
+          "taskset_name"_a, "link_markerset_dict"_a)
+      .def(
+          "set_task_linkset_points", &MarkerSets::setTaskLinkSetPoints,
+          R"(Set the points in all the MarkerSets of the specified LinkSet, in the
+          specified TaskSet, to the given dictionary, keyed by each MarkerSet's name
+          and referencing a list of 3d points.)",
+          "taskset_name"_a, "linkset_name"_a, "markerset_dict"_a)
+      .def(
+          "set_task_link_markerset_points",
+          &MarkerSets::setTaskLinkMarkerSetPoints,
+          R"(Set the marker points for the specified TaskSet's specified LinkSet's
+          specified MarkerSet to the given list of 3d points)",
+          "taskset_name"_a, "linkset_name"_a, "markerset_name"_a,
+          "marker_list"_a)
+      .def(
+          "set_all_points", &MarkerSets::setAllMarkerPoints,
```

Comment:
Not sure exactly what you mean, but this will create all the necessary components (taskSet, levelSet, markerSet) needed to save the points if they are missing.